pennylane/clusters.py

```diff
@@ -0,0 +1,476 @@
+# Copyright 2018-2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains high-level QNode processing functions and classes.
+"""
+# pylint: disable=too-many-arguments,import-outside-toplevel
+from collections.abc import Sequence
+
+from pennylane.qnodes import QNode
+from pennylane.measure import expval, var, sample
+from pennylane.operation import Observable
+
+
+MEASURE_MAP = {"expval": expval, "var": var, "sample": sample}
+
+
+def map(template, observables, device, measure="expval", interface="autograd", diff_method="best"):
+    """Map a quantum template over a list of observables to create
```

Comment:
For the docs along the lines of ``qml_cluster.rst`` seems to be missing with comparable content to:
```
qml.cluster
===========

.. currentmodule:: pennylane.cluster

.. automodapi:: pennylane.cluster
    :no-heading:
    :include-all-objects:
    :no-inheritance-diagram:
```

Reply:
Ah, this is deliberate. This is because the docs are generated in import order, and the functions and classes in this module are all imported into `__init__.py`. So in the documentation, they appear on the https://pennylane.readthedocs.io/en/stable/code/qml.html page